gap/Tools.gi

```diff
@@ -90,6 +90,55 @@ InstallMethod( BasisOfSolutionsOfHomogeneousDoubleLinearSystemInLinearCategory,
     
 end );
 
+##
+InstallMethod( BasisOfSolutionsOfHomogeneousDoubleLinearSystemInLinearCategory,
+               [ IsList, IsList ],
+               
+  function( alpha, delta )
+    local cat, zero, beta, gamma, i;
+    
+    cat := CapCategory( delta[ 1, 1 ] );
+    
+    zero := Zero( CommutativeRingOfLinearCategory( cat ) );
+    
+    if not CanCompute( cat, "BasisOfSolutionsOfHomogeneousDoubleLinearSystemInLinearCategory" ) then
+      TryNextMethod( );
+    fi;
+    
+    beta := [ ];
+    
+    gamma := [ ];
+    
+    for i in [ 1 .. Length( alpha ) ] do
+      
+      Add( beta, List( [ 1 .. Length( delta[i] ) ],
+        function( j )
+          
+          if IsEndomorphism( delta[i][j] ) then # ?and not IsIdenticalToZeroMorphism( alpha[i][j] ) then
```

Comment:
```suggestion
          if IsEndomorphism( delta_ij ) and not IsZero( alpha_ij ) then
```

Reply:
`IsZero` method could still be expensive in the category. This is why I used `IsIdenticalToZeroMorphism`, because the latest only checks whether the morphism has the same data as the zero morphism. If you still prefer to use it, just let me know.

Reply:
I agree.